### PR TITLE
Issue/0013: Combining different matchers with the ExceptionMatcher

### PR DESCRIPTION
**Pull request by oswaldobapvicjr**

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -1,3 +1,22 @@
...
+                                 junit-utils
+                                version 1.2.0
+
+                                RELEASE NOTES
+
+-----------------------------------------------------------------------------
+
+New features/enhancements:
+
+  > issue/0013: Accept different class matchers in combination with the
```
Comment (oswaldobapvicjr) [inline RELEASE-NOTES.txt:12]: Combining different matchers with the ExceptionMatcher

File: src/main/java/net/obvj/junit/utils/matchers/ExceptionMatcher.java [hunk 2]
```diff
@@ -42,16 +40,95 @@
  */
 public class ExceptionMatcher extends TypeSafeDiagnosingMatcher<Runnable>
 {
+    private enum MessageMatchingStrategy
+    {
+        EXPECTED_SUBSTRINGS
```
Comment (oswaldobapvicjr) [inline src/main/java/net/obvj/junit/utils/matchers/ExceptionMatcher.java:45]: Add Javadoc

File: src/test/java/net/obvj/junit/utils/matchers/ExceptionMatcherTest.java [hunk 3]
```diff
@@ -130,6 +134,36 @@ public void withCause_nullButHasCause_fails()
...
+    }
+
+    @Test(expected = AssertionError.class)
+    public void withMessage_combinedMatcherAndStringNotMatching_fails()
+    {
+        assertThat(() -> RUNNABLE_THROWS_ISE_WITH_MESSAGE_AND_NO_CAUSE.run(),
+                throwsException(IllegalStateException.class)
+                        .withMessage(either(startsWith(MESSAGE1)).or(containsAny(MESSAGE2).ignoreCase())));
+    }
+
```
Comment (oswaldobapvicjr) [inline src/test/java/net/obvj/junit/utils/matchers/ExceptionMatcherTest.java:166]: Add more tests to cover the scenario of passing a null matcher reference to the ExceptionMatcher (both for an exception with an actual message and an exception with no message

File: src/main/java/net/obvj/junit/utils/matchers/ExceptionMatcher.java [hunk 4]
```diff
@@ -174,6 +251,49 @@ public ExceptionMatcher withMessageContaining(String... substrings)
...
+     * }
+     * </pre>
+     *
+     * </li>
+     * </ul>
+     *
+     * @param matcher the matcher to be used in combination for exception message validation
+     * @return the matcher, incremented with the specified matcher for testing
+     */
+    public ExceptionMatcher withMessage(Matcher<String> matcher)
```
Comment (oswaldobapvicjr) [inline src/main/java/net/obvj/junit/utils/matchers/ExceptionMatcher.java:292]: Add also a `withMessage(String string)` for more testing possibilities.

File: src/test/java/net/obvj/junit/utils/matchers/ExceptionMatcherTest.java [hunk 5]
```diff
@@ -164,7 +164,7 @@ public void withMessage_combinedMatcherAndStringNotMatching_fails()
                         .withMessage(either(startsWith(MESSAGE1)).or(containsAny(MESSAGE2).ignoreCase())));
     }
 
-    @Test
+    @Test(expected = AssertionError.class)
     public void withMessage_startsWithButNoMessage_suceeds()
```
Comment (oswaldobapvicjr) [inline src/test/java/net/obvj/junit/utils/matchers/ExceptionMatcherTest.java:168]:
```suggestion
    public void withMessage_startsWithButNoMessage_fails()
```
